src/components/primitives/Forms/AddressInput/AddressInput.tsx

```diff
@@ -0,0 +1,71 @@
+import { Input } from 'components/primitives/Forms/Input';
+import useENSAvatar from 'hooks/Guilds/ens/useENSAvatar';
+import { Avatar } from 'components/Avatar';
+import { isAddress, MAINNET_ID } from 'utils';
+import { FiX } from 'react-icons/fi';
+import useENS from 'hooks/Guilds/ens/useENS';
+import { useEffect, useState } from 'react';
+import { ClickableIcon } from './AddressInput.styled';
+import { AddressInputProps } from './types';
+
+export const AddressInput: React.FC<AddressInputProps> = ({
+  value,
+  onChange,
+  isInvalid,
+  disabled = false,
+  enableENSName = true,
+  ...rest
+}) => {
+  const { imageUrl } = useENSAvatar(value, MAINNET_ID);
+  const shouldShowAvatar = !!isAddress(value);
+
+  const { name: parsedName } = useENS(value);
+
+  const [localValue, setLocalValue] = useState(
+    (enableENSName && parsedName) || value
+  );
+  const { address: addressFromLocalValue } = useENS(localValue);
+
+  useEffect(() => {
+    if (enableENSName && addressFromLocalValue) onChange(addressFromLocalValue);
+    else onChange(localValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps

```

Comment:
I guess this is fine, but we could also do: 

const `_onChange = useCallback(() => onChange(localValue))`. This way we don't need to `eslint-disable` and risk that something is not rendering. But just a minor thing :)

Reply:
I think just adding `onChange` to the dependency array should work.